### PR TITLE
doc(Lifecycle events): Document the call order

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/angular2/src/core/annotations_impl/annotations.ts [hunk 1]
```diff
@@ -1093,6 +1093,12 @@ export const onDestroy = CONST_EXPR(new LifecycleEvent("onDestroy"));
  *
  * It is invoked only if at least one of the directive's bindings has changed.
  *
+ * Lifecycle events are guaranteed to be called in the following order:
```
Comment (tbosch) [inline modules/angular2/src/core/annotations_impl/annotations.ts]: Maybe `LifecycleEvent` should become an enum. 
Also, I would vote for moving the order docs to `LifecycleEvent` instead or repeating it at every value.

**Comment by vicb** [comment]
@tbosch as of today `LifeCycleEvent` could not be an enum due to different handling in Dart and JS (number in JS, object in Dart - there is no solution to serialize this yet).

Should we merge this and revisit later or wait for a solution (I vote for 1).

**Comment by vicb** [comment]
Updated, @tbosch PTAL.

**Comment by vicb** [comment]
landed as 7648bb8ee3095358e43a50fec9ca9d1f765eeee6

**Pull request closed without merging**